src/Symfony/Component/Routing/Generator/CompiledUrlGenerator.php

```diff
@@ -50,7 +50,11 @@ public function generate(string $name, array $parameters = [], int $referenceTyp
             throw new RouteNotFoundException(sprintf('Unable to generate a URL for the named route "%s" as such route does not exist.', $name));
         }
 
-        [$variables, $defaults, $requirements, $tokens, $hostTokens, $requiredSchemes] = $this->compiledRoutes[$name];
+        [$variables, $defaults, $requirements, $tokens, $hostTokens, $requiredSchemes, $deprecations] = $this->compiledRoutes[$name];
```

Comment:
it would be safer to add ` + [6 => []]` at the end of the expression in case the compiled routes come from a previous version of Routing. This might help with upgrading.